tools/eMMC/functions.sh

```diff
@@ -254,140 +278,166 @@ dd_bootloader () {
-    touch /tmp/rootfs/etc/ssh/ssh.regenerate
+    touch ${tmp_rootfs_dir}/etc/ssh/ssh.regenerate
     flush_cache
   fi
 
-  mkdir -p /tmp/rootfs/lib/modules/$(uname -r)/ || true
 
-  message="Copying: Kernel modules" ; broadcast
-  message="rsync: /lib/modules/$(uname -r)/ -> /tmp/rootfs/lib/modules/$(uname -r)/" ; broadcast
-  rsync -aAx /lib/modules/$(uname -r)/* /tmp/rootfs/lib/modules/$(uname -r)/ || write_failure
+  echo_broadcast "==> Copying: Kernel modules"
+  echo_broadcast "===> Creating directory for modules"
+  mkdir -p ${tmp_rootfs_dir}/lib/modules/$(uname -r)/ || true
+  echo_broadcast "===> rsync: /lib/modules/$(uname -r)/ -> ${tmp_rootfs_dir}/lib/modules/$(uname -r)/"
+  rsync -aAx /lib/modules/$(uname -r)/* ${tmp_rootfs_dir}/lib/modules/$(uname -r)/ || write_failure
   flush_cache
 
-  message="Copying: ${source}p${media_rootfs} -> ${destination}p${media_rootfs} complete" ; broadcast
-  message="-----------------------------" ; broadcast
-
-  message="Final System Tweaks:" ; broadcast
+  echo_broadcast "Copying: ${source}p${media_rootfs} -> ${destination}p${media_rootfs} complete"
 
+  echo_broadcast "Final System Tweaks:"
+  echo_broadcast "==> Put root uuid in uEnv.txt"
   unset root_uuid
   root_uuid=$(/sbin/blkid -c /dev/null -s UUID -o value ${destination}p${media_rootfs})
   if [ "${root_uuid}" ] ; then
-    sed -i -e 's:uuid=:#uuid=:g' /tmp/rootfs/boot/uEnv.txt
-    echo "uuid=${root_uuid}" >> /tmp/rootfs/boot/uEnv.txt
+    sed -i -e 's:uuid=:#uuid=:g' ${tmp_rootfs_dir}/boot/uEnv.txt
+    echo "uuid=${root_uuid}" >> ${tmp_rootfs_dir}/boot/uEnv.txt
 
-    message="UUID=${root_uuid}" ; broadcast
+    echo_broadcast "===> UUID=${root_uuid}"
     root_uuid="UUID=${root_uuid}"
   else
-    #really a failure...
+    #FIXME: really a failure...
     root_uuid="${source}p${media_rootfs}"
   fi
 
-  message="Generating: /etc/fstab" ; broadcast
-  echo "# /etc/fstab: static file system information." > /tmp/rootfs/etc/fstab
-  echo "#" >> /tmp/rootfs/etc/fstab
-  echo "${root_uuid}  /  ext4  noatime,errors=remount-ro  0  1" >> /tmp/rootfs/etc/fstab
-  echo "debugfs  /sys/kernel/debug  debugfs  defaults  0  0" >> /tmp/rootfs/etc/fstab
-  cat /tmp/rootfs/etc/fstab
-
-  message="/boot/uEnv.txt: disabling eMMC flasher script" ; broadcast
-  sed -i -e 's:'$emmcscript':#'$emmcscript':g' /tmp/rootfs/boot/uEnv.txt
-  cat /tmp/rootfs/boot/uEnv.txt
-  message="-----------------------------" ; broadcast
-
+  echo_broadcast "==> Generating: /etc/fstab"
+  echo "# /etc/fstab: static file system information." > ${tmp_rootfs_dir}/etc/fstab
+  echo "#" >> ${tmp_rootfs_dir}/etc/fstab
+  echo "${root_uuid}  /  ext4  noatime,errors=remount-ro  0  1" >> ${tmp_rootfs_dir}/etc/fstab
+  echo "debugfs  /sys/kernel/debug  debugfs  defaults  0  0" >> ${tmp_rootfs_dir}/etc/fstab
+  cat ${tmp_rootfs_dir}/etc/fstab
+
+  #FIXME: What about when you boot from a fat partition /boot ?
+  echo_broadcast "==> /boot/uEnv.txt: disabling eMMC flasher script"
```

Comment:
That's actually where the "v3" in file names come from..  With v3 it's "assumed" to read /boot/uEnv.txt  but there is a wrapper we setup for the offical images, to deal with really old eMMC bootloaders aka: 1st partition: /uEnv.txt -> 2nd partition /boot/uEnv.txt..  However once you flash an eMMC with this script that old /uEnv.txt is no longer required as you just updated the eMMC bootloader..